src/Simulation/Simulators/QuantumSimulator/Dump.cs

```diff
@@ -26,19 +26,9 @@ public uint[] QubitIds
             }
         }
 
-        protected override QVoid process(Action<string> channel, IQArray<Qubit>? qubits)
+        public override uint[] GetQubitIds()
```

Comment:
I'd suggest an expression-body method here:

```csharp
public override uint[] GetQubitIds() =>
    QubitIds;
```

More to the point, though, it looks like this method duplicates `QubitIds { get; }`. Would it be better to make the abstract class define `QubitIds { get; }` directly?

Reply:
Restructured a bit. Thanks!